cmake/CompilerWarnings.cmake

```diff
@@ -40,12 +41,12 @@ function(set_default_compiler_warnings target_name)
         -Wimplicit-fallthrough)
 
     if (OPENASSETIO_WARNINGS_AS_ERRORS)
-        set(CLANG_WARNINGS ${CLANG_WARNINGS} -Werror)
-        set(MSVC_WARNINGS ${MSVC_WARNINGS} /WX)
+        set(clang_warnings ${clang_warnings} -Werror)
+        set(vs_msvc_warnings ${vs_msvc_warnings} /WX)
```

Comment:
I know this is an awkward name, but `MSVC_WARNINGS` and `MSVCPP_WARNINGS` are both built in symbols, so we were getting complaints about overlap.

Reply:
Interesting, already a good catch by the linter!